packages/indy-vdr/src/IndyVdrApi.ts

```diff
@@ -0,0 +1,71 @@
+import type { Key } from '@aries-framework/core'
+import type { EndorseDidTxAction } from '@aries-framework/core/src/modules/dids/types'
+import type { AttribRequest, IndyVdrRequest } from '@hyperledger/indy-vdr-shared'
+
+import { AgentContext, injectable, TypedArrayEncoder } from '@aries-framework/core'
+
+import { verificationKeyForIndyDid, parseIndyDid } from './dids/didIndyUtil'
+
+@injectable()
+export class IndyVdrApi {
+  private agentContext: AgentContext
+
+  public constructor(agentContext: AgentContext) {
+    this.agentContext = agentContext
+  }
+
+  private async multiSignRequest<Request extends IndyVdrRequest>(
+    request: Request,
+    signingKey: Key,
+    identifier: string
+  ) {
+    const signature = await this.agentContext.wallet.sign({
+      data: TypedArrayEncoder.fromString(request.signatureInput),
+      key: signingKey,
+    })
+
+    request.setMultiSignature({
+      signature,
+      identifier,
+    })
+
+    return request
+  }
+
+  private async signRequest<Request extends IndyVdrRequest>(request: Request, signingKey: Key) {
+    const signature = await this.agentContext.wallet.sign({
+      data: TypedArrayEncoder.fromString(request.signatureInput),
+      key: signingKey,
+    })
+
+    request.setSignature({
+      signature,
+    })
+
+    return request
+  }
+
+  public async endorseTransaction<Request extends IndyVdrRequest>(transaction: Request, endorserDid: string) {
```

Comment:
I am not sure about this. How can we later create a Nym / AttribRequest from the JSON representation?

Reply:
I think you can just do `request.body` to get a json/string representation of the request

Reply:
Then on the other side you can do `new CustomRequest({ customRequest: theRequestJsonOrString })`. 

It may be that the types aren't fully happy yet. In that case if a cast works, that's fine for now and we can update indy-vdr to allow more loose types (just any json / string)